stl/inc/mutex

```diff
@@ -508,61 +508,70 @@ public:
-    return _Invoke_stored_explicit(_STD move(_Tuple), index_sequence_for<_Types...>{});
-}
-
 // FUNCTION TEMPLATE call_once
-[[noreturn]] _CRTIMP2_PURE void __CLRCALL_PURE_OR_CDECL _XGetLastError();
-
-template <class _Tuple, class _Seq, size_t _Idx>
-int __stdcall _Callback_once(void*, void* _Pv, void**) { // adapt call_once() to callback API
-    _Tuple* _Ptup = static_cast<_Tuple*>(_Pv);
-
-    _TRY_BEGIN
-    // Note explicit _Seq{} selects every element from *_Ptup except the last,
-    // which contains call_once's exception_ptr.
-    _Invoke_stored_explicit(_STD move(*_Ptup), _Seq{});
-    _CATCH_ALL
-    auto& _Ref = _STD get<_Idx>(*_Ptup);
-    _Ref       = _STD current_exception();
-    return 0;
-    _CATCH_END
+#ifdef _M_CEE
+#define _WINDOWS_API              __stdcall
+#define _RENAME_WINDOWS_API(_Api) _Api##_clr
+#else // ^^^ _M_CEE // !_M_CEE vvv
+#define _WINDOWS_API              __declspec(dllimport) __stdcall
+#define _RENAME_WINDOWS_API(_Api) _Api
+#endif // _M_CEE
 
-    return 1;
-}
+// WINBASEAPI
+// BOOL
+// WINAPI
+// InitOnceBeginInitialize(
+//     _Inout_ LPINIT_ONCE lpInitOnce,
+//     _In_ DWORD dwFlags,
+//     _Out_ PBOOL fPending,
+//     _Outptr_opt_result_maybenull_ LPVOID* lpContext
+//     );
+extern "C" int _WINDOWS_API _RENAME_WINDOWS_API(__std_init_once_begin_initialize)(
+    void** _LpInitOnce, unsigned long _DwFlags, int* _FPending, void** _LpContext) noexcept;
+// WINBASEAPI
+// BOOL
+// WINAPI
+// InitOnceComplete(
+//     _Inout_ LPINIT_ONCE lpInitOnce,
+//     _In_ DWORD dwFlags,
+//     _In_opt_ LPVOID lpContext
+//     );
+extern "C" int _WINDOWS_API _RENAME_WINDOWS_API(__std_init_once_complete)(
+    void** _LpInitOnce, unsigned long _DwFlags, void* _LpContext) noexcept;
+// #define RTL_RUN_ONCE_INIT_FAILED    0x00000004UL
+// #define INIT_ONCE_INIT_FAILED       RTL_RUN_ONCE_INIT_FAILED
+_INLINE_VAR constexpr unsigned long _Init_once_init_failed = 0x4UL;
+struct _Init_once_completer {
+    once_flag& _Once;
+    unsigned long _DwFlags;
+    ~_Init_once_completer() {
+        if (_RENAME_WINDOWS_API(__std_init_once_complete)(&_Once._Opaque, _DwFlags, nullptr) == 0) {
+            _CSTD abort();
+        }
+    }
+};
 
 template <class _Fn, class... _Args>
-void(call_once)(once_flag& _Flag, _Fn&& _Fx, _Args&&... _Ax) { // call _Fx(_Ax...) once
-    using _Tuple = tuple<_Fn&&, _Args&&..., exception_ptr&>;
-    using _Seq   = make_index_sequence<1 + sizeof...(_Args)>;
-
-    exception_ptr _Exc;
-    _Tuple _Tup(_STD forward<_Fn>(_Fx), _STD forward<_Args>(_Ax)..., _Exc);
-
-    _Execute_once_fp_t _Fp = &_Callback_once<_Tuple, _Seq, 1 + sizeof...(_Args)>;
-
-    if (_Execute_once(_Flag, _Fp, _STD addressof(_Tup)) != 0) {
-        return;
+void(call_once)(once_flag& _Once, _Fn&& _Fx, _Args&&... _Ax) noexcept(
+    noexcept(_STD invoke(_STD forward<_Fn>(_Fx), _STD forward<_Args>(_Ax)...))) // Strengthened
+{

```

Comment:
Nitpicks: We conventionally use lowercase here, and we prefer to avoid putting the opening brace on the next line.
```suggestion
    noexcept(_STD invoke(_STD forward<_Fn>(_Fx), _STD forward<_Args>(_Ax)...))) /* strengthened */ {
```